File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/ui/AdvancedMappingControlAdapter.java
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.ui;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.dolphinemu.dolphinemu.databinding.ListItemAdvancedMappingControlBinding;

import java.util.function.Consumer;

public final class AdvancedMappingControlAdapter
        extends RecyclerView.Adapter<AdvancedMappingControlViewHolder>
{
  private final Consumer<String> mOnClickCallback;

  private String[] mControls = new String[0];

  public AdvancedMappingControlAdapter(Consumer<String> onClickCallback)
  {
    mOnClickCallback = onClickCallback;
  }

  @NonNull @Override
  public AdvancedMappingControlViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
          int viewType)
  {
    LayoutInflater inflater = LayoutInflater.from(parent.getContext());

    ListItemAdvancedMappingControlBinding binding =
            ListItemAdvancedMappingControlBinding.inflate(inflater);
    return new AdvancedMappingControlViewHolder(binding, mOnClickCallback);
  }

  @Override
  public void onBindViewHolder(@NonNull AdvancedMappingControlViewHolder holder, int position)
  {
    holder.bind(mControls[position]);
  }

  @Override
  public int getItemCount()
  {
    return mControls.length;
  }

  public void setControls(String[] controls)
  {
    mControls = controls;
    notifyDataSetChanged();
  }
}

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/input/ui/AdvancedMappingControlAdapter.kt
// SPDX-License-Identifier: GPL-2.0-or-later

package org.dolphinemu.dolphinemu.features.input.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.lifecycle.Lifecycle
import androidx.recyclerview.widget.RecyclerView
import org.dolphinemu.dolphinemu.databinding.ListItemAdvancedMappingControlBinding
import org.dolphinemu.dolphinemu.features.input.model.CoreDevice
import java.util.function.Consumer

class AdvancedMappingControlAdapter(
    private val parentLifecycle: Lifecycle,
    private val onClickCallback: Consumer<String>
) : RecyclerView.Adapter<AdvancedMappingControlViewHolder>() {

    private var controls = emptyArray<CoreDevice.Control>()

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): AdvancedMappingControlViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding = ListItemAdvancedMappingControlBinding.inflate(inflater)
        return AdvancedMappingControlViewHolder(binding, parentLifecycle, onClickCallback)
    }

    override fun onBindViewHolder(holder: AdvancedMappingControlViewHolder, position: Int) =
        holder.bind(controls[position])

    override fun getItemCount(): Int = controls.size

    fun setControls(controls: Array<CoreDevice.Control>) {
        this.controls = controls
        notifyDataSetChanged()
    }

    override fun onViewRecycled(holder: AdvancedMappingControlViewHolder) {
        super.onViewRecycled(holder)
        holder.onViewRecycled()
    }

    override fun onViewAttachedToWindow(holder: AdvancedMappingControlViewHolder) {
        super.onViewAttachedToWindow(holder)
        holder.onViewAttachedToWindow()
    }

    override fun onViewDetachedFromWindow(holder: AdvancedMappingControlViewHolder) {
        super.onViewDetachedFromWindow(holder)
        holder.onViewDetachedFromWindow()
    }
}
